config/presets/testnet.go

```diff
@@ -1,73 +1,145 @@
+	}
+	defaultdir := filepath.Join(home, "spacemesh-testnet", "/")
+	return config.Config{
+		BaseConfig: config.BaseConfig{
+			DataDirParent:       defaultdir,
+			FileLock:            filepath.Join(os.TempDir(), "spacemesh.lock"),
+			MetricsPort:         1010,
+			DatabaseConnections: 16,
+			NetworkHRP:          "smtest",
 
-	conf.POST.K1 = 26
-	conf.POST.K2 = 37
-	conf.POST.K3 = 37
-	conf.POST.LabelsPerUnit = 20 * 1024 / 16 // 20 kB units
-	conf.POST.MaxNumUnits = 4
-	conf.POST.MinNumUnits = 2
+			LayerDuration:  5 * time.Minute,
+			LayerAvgSize:   50,
+			LayersPerEpoch: 288,
 
-	conf.SMESHING.CoinbaseAccount = types.GenerateAddress([]byte("1")).String()
-	conf.SMESHING.Start = false
-	conf.SMESHING.Opts.ProviderID.SetInt64(int64(initialization.CPUProviderID()))
-	conf.SMESHING.Opts.NumUnits = 2
-	conf.SMESHING.Opts.Throttle = true
+			TxsPerProposal: 700,       // https://github.com/spacemeshos/go-spacemesh/issues/4559
+			BlockGasLimit:  100107000, // 3000 of spends
 
-	conf.Beacon.FirstVotingRoundDuration = 3 * time.Minute
-	conf.Beacon.GracePeriodDuration = 10 * time.Second
-	conf.Beacon.ProposalDuration = 30 * time.Second
-	conf.Beacon.RoundsNumber = 6
-	conf.Beacon.BeaconSyncWeightUnits = 30
-	conf.Beacon.VotesLimit = 100
-	conf.Beacon.VotingRoundDuration = 50 * time.Second
-	conf.Beacon.WeakCoinRoundDuration = 10 * time.Second
+			OptFilterThreshold: 90,
 
-	return conf
+			TickSize:    666514,
+			PoETServers: []string{},
+		},
+		Genesis: &config.GenesisConfig{
+			GenesisTime: "2023-09-13T08:00:00Z",
+			ExtraData:   "0000000000000000000000c76c58ebac180989673fd6d237b40e66ed5c976ec3",
+		},
+		Tortoise: tortoise.Config{
+			Hdist:                    10,
+			Zdist:                    2,
+			WindowSize:               10000,
+			MaxExceptions:            1000,
+			BadBeaconVoteDelayLayers: 4032,
+			// 100 - is assumed minimal number of units
+			// 100 - half of the expected poet ticks
+			MinimalActiveSetWeight: 100 * 100,
+		},
+		HARE: hareConfig.Config{
+			N:               200,
+			ExpectedLeaders: 5,
+			RoundDuration:   25 * time.Second,
+			WakeupDelta:     25 * time.Second,
+			LimitConcurrent: 2,
+			LimitIterations: 4,
+		},
+		HareEligibility: eligConfig.Config{
+			ConfidenceParam: 20,
+		},
+		Beacon: beacon.Config{
+			Kappa:                    40,
+			Q:                        big.NewRat(1, 3),
+			Theta:                    big.NewRat(1, 4),
+			GracePeriodDuration:      10 * time.Minute,
+			ProposalDuration:         4 * time.Minute,
+			FirstVotingRoundDuration: 30 * time.Minute,
+			RoundsNumber:             20,
+			VotingRoundDuration:      4 * time.Minute,
+			WeakCoinRoundDuration:    4 * time.Minute,
+			VotesLimit:               100,
+			BeaconSyncWeightUnits:    800,
+		},
+		POET: activation.PoetConfig{
+			PhaseShift:        12 * time.Hour,
+			CycleGap:          2 * time.Hour,
+			GracePeriod:       10 * time.Minute,
+			RequestRetryDelay: 10 * time.Second,
+			MaxRequestRetries: 10,
```

Comment:
The total request retry time in case of 10 retries could be bigger than the grace period.

```
min_total_retry_time = (10 + 10*10) / 2 * 10 = 550s
max_total_retry_time = 2* min_total_retry_time = 1100s
```
because we set max retry to `2*RequestRetryDelay`.

see https://github.com/hashicorp/go-retryablehttp/blob/571a88bc9c3b7c64575f0e9b0f646af1510f2c76/client.go#L539-L575

```suggestion
			GracePeriod:       10 * time.Minute,
			RequestRetryDelay: 5 * time.Second,
			MaxRequestRetries: 10,
```

Reply:
can you tell me what i should set?